docs/development/local_setup.md

```diff
@@ -1,65 +1,101 @@
-# Preparing the Local Development Setup (Mac OS X)
+# Preparing the setup
 
 Conceptionally, the Gardener is designated to run in containers within a Pod inside an Kubernetes cluster. It extends the API via the user-aggregated API server concepts. However, if you want to develop it, you may want to work locally with the Gardener without building a Docker image and deploying it to a cluster each and every time. That means that the Gardener runs outside a Kubernetes cluster which requires providing a [Kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/authenticate-across-clusters-kubeconfig/) in your local filesystem and point the Gardener to it when starting it (see below).
 
+Further details could be found in
+1. [principles of Kubernetes](https://kubernetes.io/docs/concepts/), and its [components](https://kubernetes.io/docs/concepts/overview/components/)
+1. [architecture of the Garden](https://github.com/gardener/documentation/wiki/Architecture)
+
+
+This setup is based on [minikube](https://github.com/kubernetes/minikube),  a Kubernetes cluster running on a single node.
+
 ## Installing Golang environment
-Install the latest version of Golang (at least `v1.9.2` is required) by using [Homebrew](https://brew.sh/):
+Install the latest version of Golang (at least `v1.9.2` is required). For Mac OS you could use [Homebrew](https://brew.sh/):
 
 ```bash
 $ brew install golang
 ```
+For other OS, please check [Go installation documentation](https://golang.org/doc/install).
 
 Make sure to set your `$GOPATH` environment variable properly (conventionally, it points to `$HOME/go`).
 
 For your convenience, you can add the `bin` directory of the `$GOPATH` to your `$PATH`: `PATH=$PATH:$GOPATH/bin`, but it is not necessarily required.
 
+We use [Dep](https://github.com/golang/dep) for managing Golang package dependencies. Please install it
+on Mac OS via
+
+```bash
+$ brew install dep
+```
+
+On other OS please check the [Dep installation documentation](https://golang.github.io/dep/docs/installation.html) and the [Dep releases page](https://github.com/golang/dep/releases). After downloading the appropriate release in your `$GOPATH/bin` folder you need to make it executable via `chmod +x <dep-release>` and to rename it to dep via `mv dep-<release> dep`.
+
+
+### [Optional] Golint
```

Comment:
Why is that marked `[Optional]`? Nobody should commit anything without running at least `make verify`.